src/modules/lost/utilities.c

```diff
@@ -213,20 +231,55 @@ void lost_free_loc(p_loc_t ptr)
 		pkg_free(ptr->profile);
 
 	pkg_free(ptr);
-	ptr = NULL;
+	*loc = NULL;
 }
 
 /*
  * lost_free_loc(ptr)
  * frees a held location request object
  */
-void lost_free_held(p_held_t ptr)
+void lost_free_held(p_held_t *held)
 {
-	pkg_free(ptr->identity);
-	pkg_free(ptr->type);
+	p_held_t ptr;
+
+	if(held == NULL)
+		return;
+
+	ptr = *held;
+
+	if(ptr->identity)
+		pkg_free(ptr->identity);
+	if(ptr->type)
+		pkg_free(ptr->type);
 
 	pkg_free(ptr);
-	ptr = NULL;
+	*held = NULL;
+}
+
+/*
+ * lost_copy_string(str, int*) {
```

Comment:
Have you considered using the existing ut.h pkg_str_dup(..) function?

Reply:
yes true `pkg_str_dup(..)` would be an option; basically I only use the function when a pointer to a string is necessary and requires zero termination